src/jwt/decode.ts

```diff
@@ -0,0 +1,14 @@
+import { decodeFromBase64Url } from "./utils";
+import type { JWTRegisteredClaims } from "./types";
+
+export function decodeJWT<T extends Record<string, any> = Record<string, any>>(
+  token: string
+) {
+  const parts = token.split(".");
+  if (parts.length !== 3) {
+    throw new Error("Invalid token format");
+  }
+
+  const [, payload] = parts;
+  return JSON.parse(decodeFromBase64Url(payload)) as JWTRegisteredClaims & T;
```

Comment:
Decode / Encode can fail for many reasons (runtime errors). Would we nice if create internal `_decodeJWT` util and export a wrapper with `try/catch` protection. I am open to an _unsafe_ version which can throws more explicit errors but as general practice across UnJS, we tend to do similar approach first.

Reply:
Good idea! To keep logic close to UnJS, I'd prefer a `{ strict: boolean }` option, just like `readBody` from h3. What do you think?

Also: What about using destr instead of try/catch and JSON.parse? Would that be OK or do you intend to keep uncrypto deps-free?

Reply:
`try/catch` done!